### PR TITLE
use generic oidc interface on client

### PR DESCRIPTION
**Pull request by DallanQ**
also make client and server work with not-quite-compliant Cognito by changing just a few environment variables.

Jim, two things I'm concerned about:

1. The Cognito domain cannot end in a slash, so I added the https:// and the trailing slash to the environment variable for auth0 and took it out of the code. I updated the environment variable in docker-compose and in deploy/aswlambda/cms.cf.yaml. Is there anything else I need to do?

2. I had to add a new html file: uglyui/public/callback.html. This file should be deployed to the s3 bucket and served for https://cms.ourroots.org/callback.html. How do I do that?

### CODE REVIEW

**Comment by jancona** [comment]
Answering your questions:

1. That should work for `OIDC_DOMAIN`.
2. You shouldn't have to do anything to make the new file available on the master branch. The dynamo branch has my deploy changes to route almost everything to `index.html`, so I will have to special-case `callback.html` there.

File: api/api.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 
 	"github.com/cenkalti/backoff/v4"
 
-	"github.com/coreos/go-oidc"
+	"github.com/ourrootsorg/go-oidc"
```
Comment (jancona) [inline api/api.go:16]: The plan is to switch that back to the `coreos` version once they accept your PR, correct?

Comment (DallanQ) [reply api/api.go:16]: That's right.

**Comment by DallanQ** [comment]
I'll go ahead and merge the PR. At some point could you change the deploy to special-case redirecting callback.html to callback.html?

**Comment by jancona** [comment]
I'll do that when I merge it to the dynamo branch.

**Pull request merged**